File: client/unity/Assets/Scripts/Assembly-CSharp/AvMain.cs
public class AvMain
{
	public static int hd;

	public static int hDuBox;

	public static int duPopup;

	public static int hFillTab;

	public static int hCmd;

	public static sbyte hBlack;

	public static sbyte hBorder;

	public static sbyte hNormal;

	public static sbyte hSmall;

	public static sbyte hBlack2;

	public static float zoom;

	public static byte indexLeft;

	public static byte indexCenter;

	public static byte indexRight;

	public Command left;

	public Command center;

	public Command right;

	public bool isHide;

	public bool isTran;

	public static bool isQwerty;

	public static int lsplash;

	public static int csplash;

	public static int rsplash;

	public static string lsplashs;

	public static string csplashs;

	public static string rsplashs;

	static AvMain()
	{
		zoom = 1f;
		if (Main.hdtype == 0)
		{
			hd = 1;
		}
		if (Main.hdtype == 1)
		{
			hd = 1;
		}
		if (Main.hdtype == 2)
		{
			hd = 2;
		}
	}

	public virtual void keyPress(int keyCode)
	{
	}

	public virtual void paint(MyGraphics g)
	{
		Canvas.resetTransNotZoom(g);
		Canvas.paint.paintCmd(g, left, center, right);
	}

	public virtual void commandAction(int index)
	{
	}

	public virtual void commandActionPointer(int index, int subIndex)
	{
	}

	public virtual void commandTab(int index)
	{
	}

	public virtual void closeTabAll()
	{
	}

	public virtual void initTabTrans()
	{
	}

	private void click(Command cmd)
	{
		if (cmd != null)
		{
			Canvas.isPointerClick = false;
			Canvas.isPointerRelease = false;
			Canvas.endDlg();
			perform(cmd);
			SoundManager.playSound(7);
			indexRight = (indexCenter = (indexLeft = 0));
		}
	}

	public virtual void updateKey()
	{
		if (lsplash > 0)
		{
			lsplash--;
		}
		if (csplash > 0)
		{
			csplash--;
		}
		if (rsplash > 0)
		{
			rsplash--;
		}
		if (Canvas.currentDialog != null || Canvas.welcome == null || !Welcome.isPaintArrow || Welcome.lastScr != Canvas.currentMyScreen)
		{
			Canvas.paint.setDrawPointer(left, center, right);
			if (Canvas.currentMyScreen != MessageScr.me)
			{
				Canvas.paint.setBack();
			}
		}
		else if (Canvas.currentMyScreen != MessageScr.me && (Canvas.welcome == null || !Welcome.isPaintArrow || Welcome.lastScr != Canvas.currentMyScreen))
		{
			Canvas.paint.setBack();
		}
		if (!Canvas.isPointerRelease)
		{
			return;
		}
		switch (Canvas.paint.collisionCmdBar(left, center, right))
		{
		case 0:
			if (indexLeft == 4 || Canvas.stypeInt == 0)
			{
				click(left);
			}
			break;
		case 1:
			if (indexCenter == 4 || Canvas.stypeInt == 0)
			{
				click(center);
			}
			break;
		case 2:
			if (indexRight == 4 || Canvas.stypeInt == 0)
			{
				click(right);
			}
			break;
		}
		indexRight = (indexCenter = (indexLeft = 0));
	}

	public virtual void perform(Command cmd)
	{
		if (cmd != null)
		{
			if (cmd.action != null || cmd.ipaction != null)
			{
				cmd.perform();
			}
			else if (cmd.pointer != null)
			{
				cmd.pointer.commandActionPointer(cmd.indexMenu, cmd.subIndex);
			}
			else
			{
				commandTab(cmd.indexMenu);
			}
		}
	}
}

File: client/java/src/avt/AvMain.java
package avt;

import javax.microedition.lcdui.Graphics;
import main.Canvas;
import main.GameMidlet;

public class AvMain {
   public static int hd = 1;
   public static int hDuBox;
   public static int duPopup;
   public static int hFillTab;
   public Command left;
   public Command center;
   public Command right;
   public static byte hBlack;
   public static byte hBorder;
   public static byte hNormal;
   public static byte hSmall;
   public boolean isHide_;
   public static boolean isQwerty = false;
   private static byte a = 0;
   private static byte b = 0;
   private static byte c = 0;

   public void initCmd() {
   }

   public void keyPress(int var1) {
   }

   public void paint(Graphics var1) {
      Canvas.resetTrans(var1);
      Canvas.paint.paintCmd(var1, this.left, this.center, this.right);
   }

   public void commandActionPointer(int var1, int var2) {
   }

   public void commandActionPointer(int var1) {
   }

   public void commandTab(int var1, int var2) {
   }

   private void click(Command var1) {
      if (var1 != null) {
         Canvas.isPointerClick = false;
         Canvas.isPointerRelease = false;
         Canvas.endDlg();
         this.perform(var1);
      }

   }

   public void update() {
   }

   public void updateKey() {
      if (Canvas.isPointerRelease) {
         if (Canvas.isPointerInRect(0, Canvas.ae[0].y, Canvas.w - 1, Canvas.hTab)) {
            switch (Canvas.paint.getDisplayValue()) {
               case 0:
                  if (Canvas.stypeInt == 0) {
                     this.click(this.left);
                  }
                  break;
               case 1:
                  if (Canvas.stypeInt == 0) {
                     this.click(this.center);
                  }
                  break;
               case 2:
                  if (Canvas.stypeInt == 0) {
                     this.click(this.right);
                  }
            }
         }

         a = 0;
         b = 0;
         c = 0;
         if (Canvas.isPaintIconVir()) {
            if (Canvas.isPointer(0, 0, 50 * hd, 50 * hd)) {
               if (!OptionScr.isVirTualKey) {
                  if (TField.m) {
                     isQwerty = true;
                     TField.m = false;
                  }
               } else if (isQwerty) {
                  TField.m = true;
               }

               OptionScr.isVirTualKey = !OptionScr.isVirTualKey;
               Canvas.instance.sizeChanged(0, 0);
               Canvas.isPointerRelease = false;
            }

            if (GameMidlet.CLIENT_TYPE == 9 && Canvas.isPointer(50, 0, 50 * hd, 50 * hd)) {
               if (!OptionScr.isVirTualKey) {
                  OptionScr.isVirTualKey = true;
                  OptionScr.gI().mapFocus[4] = 1;
                  Canvas.instance.setSize();
                  Canvas.z.isPointerClick = true;
                  if (Canvas.currentMyScreen == MapScr.gI()) {
                     ChatTextField.gI().parentMyScreen = MapScr.gI();
                     ChatTextField.isShow = true;
                  }
               } else {
                  OptionScr.isVirTualKey = false;
                  OptionScr.gI().mapFocus[4] = 0;
                  Canvas.instance.setSize();
                  if (Canvas.currentMyScreen == MapScr.gI()) {
                     ChatTextField.isShow = false;
                  }
               }

               Canvas.isPointerRelease = false;
            }
         }
      }

      if (Canvas.isKeyPressed(5)) {
         if (this.center != null) {
            Canvas.endDlg();
            this.perform(this.center);
            return;
         }

         if (Canvas.menuMain == this) {
            this.perform(this.left);
            return;
         }
      } else {
         if (Canvas.isKeyPressed(12)) {
            this.perform(this.left);
            return;
         }

         if (Canvas.E) {
            if (Canvas.isKeyPressed(13) || Canvas.keyReleased[13]) {
               Canvas.keyReleased[13] = false;
               this.perform(this.right);
               return;
            }
         } else if (Canvas.isKeyPressed(13)) {
            this.perform(this.right);
         }
      }

   }

   public final void perform(Command var1) {
      if (var1 != null) {
         if (var1.action != null) {
            var1.action.perform();
            return;
         }

         if (var1.pointer != null) {
            var1.pointer.commandActionPointer(var1.indexMenu);
            return;
         }

         if (ChatTextField.isShow) {
            ChatTextField.gI().commandTab(var1.indexMenu, var1.subIndex);
            return;
         }

         this.commandTab(var1.indexMenu, var1.subIndex);
      }

   }
}
